scale/job/seed/metadata.py

```diff
@@ -17,53 +19,77 @@
     METADATA_SCHEMA = json.load(schema_file)
 METADATA_SUFFIX = '.metadata.json'
 
+UNSUPPORTED_TYPES = ('FeatureCollection', 'GeometryCollection')
+
 
 class SeedMetadata(object):
-    """Represents the extended metadata for a single file captured from a Seed job"""
+    """Represents the extended metadata for a single file in Seed job.
+
+    Object instantiation from existing JSON should be accomplished via the metadata_from json helper.
+    This will safeguard us from having to deal with FeatureCollection and GeometryCollection objects."""
+
+    def __init__(self):
+        """Initialize the metadata object with a basic placeholder"""
+
+        self._data = {
+            'type': 'Feature',
+            'geometry': None,
+            'properties': None
+        }
 
-    def __init__(self, definition, do_validate=True):
-        """Creates a metadata class from the given definition. If the definition is invalid, a
-        :class:`job.configuration.interface.exceptions.InvalidMetadataDefinition` exception will be thrown.
+    @staticmethod
+    def metadata_from_json(metadata, do_validate=True):
+        """Creates a metadata object from the given definition. If the definition is invalid, a
+        :class:`job.seed.exceptions.InvalidSeedMetadataDefinition` exception will be thrown.
 
-        :param definition: The interface definition
-        :type definition: dict
+        :param metadata: The metadata definition
+        :type metadata: dict
         :param do_validate: Whether to perform validation on the JSON schema
         :type do_validate: bool
+        :raises InvalidSeedMetadataDefinition: If schema validation fails
         """
 
-        self.definition = definition
+        self = SeedMetadata()
 
         try:
             if do_validate:
-                validate(definition, METADATA_SCHEMA)
+                validate(metadata, METADATA_SCHEMA)
         except ValidationError as validation_error:
-            raise InvalidSeedMetadataDefinition('JSON_VALIDATION_ERROR', 'Error validating againsg schema: %s' % validation_error)
-
-        self._populate_default_values()
-
-    def get_geometry(self):
-        """Retrieves GeoJSON geometry if it is available in metadata object
-
-        :return: Geometry as defined by GeoJSON specification, possibly None if unset
-        :rtype: dict or None
+            raise InvalidSeedMetadataDefinition('JSON_VALIDATION_ERROR', 'Error validating against schema: %s' %
+                                                validation_error)
+        if metadata['type'] in UNSUPPORTED_TYPES:
+            raise InvalidSeedMetadataDefinition('UNSUPPORTED_GEOJSON', 'Scale does not process GeoJSON '
+                                                                       'FeatureCollection or GeometryCollection type')
```

Comment:
GeometryCollection (1) doesn't allow for us to capture metadata (no `properties` object) and (2) it is not deterministic which Geometry is associated with our product. FeatureCollection doesn't have the first problem but also suffers from the lack of determinism on the association to product. We could do something like only parse the first element of the collection and just assume that is correct, but again its just a best guess that could be wrong. Since we are using an individual side-car per product it makes the most sense to allow only a 1-1 relationship between properties and metadata.